src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -442,6 +576,64 @@ const FormCreationForm = ({csrftoken, formUuid, formHistoryUrl }) => {
         }
     };
 
+    const onRuleChange = (index, event) => {
+        const { name, value } = event.target;
+        dispatch({
+            type: 'CHANGED_RULE',
+            payload: {name, value, index},
+        });
+    };
+
+    const onRuleDelete = (index) => {
+        dispatch({
+            type: 'DELETED_RULE',
+            payload: {index: index}
+        });
+    };
+
+    const onRuleAdd = () => {
+        dispatch({type: 'ADD_RULE'});
+    };
+
+    const submitRules = async () => {
```

Comment:
Can we move this into `data.js` and let it take some variables, so that you can call `submitRules(state.rules)` from the component and catch possible exceptions, to then relay all that to the dispatch? I really want to make this component less complex.